src/wallet/wallet.cpp

```diff
@@ -1362,65 +1362,32 @@ CAmount CWalletTx::GetCredit(const isminefilter& filter) const
     return credit;
 }
 
-CAmount CWalletTx::GetImmatureCredit(bool fUseCache) const
+CAmount CWalletTx::GetImmatureCredit(const isminefilter &filter, bool fUseCache) const
 {
-    if (IsCoinBase() && GetBlocksToMaturity() > 0 && IsInMainChain())
-    {
-        if (fUseCache && fImmatureCreditCached)
-            return nImmatureCreditCached;
-        nImmatureCreditCached = pwallet->GetCredit(*this, ISMINE_SPENDABLE);
-        fImmatureCreditCached = true;
-        return nImmatureCreditCached;
-    }
-
-    return 0;
-}
-
-CAmount CWalletTx::GetAvailableCredit(bool fUseCache) const
-{
-    if (pwallet == 0)
-        return 0;
+    if (IsCoinBase() && GetBlocksToMaturity() > 0 && IsInMainChain()) {
+        CAmount credit = 0;
+        if (filter & ISMINE_SPENDABLE) {
```

Comment:
What if filter is ISMINE_ALL? I'd prefer to see either an exact equivalent check, or a switch case, or an assert to prevent unhandled cases.